llama_index/tools/ondemand_tool_loader_tool.py

```diff
@@ -0,0 +1,96 @@
+"""Ad-hoc data loader tool.
+
+Tool that wraps any other tool, and is able to load data on-demand.
+
+"""
+
+
+from llama_hub.tools.types import BaseTool, ToolMetadata
+from llama_index.readers.base import BaseReader
+from typing import Any, Optional, Dict, Type
+from llama_index.indices.base import BaseIndex
+from llama_index.indices.vector_store import VectorStoreIndex
+from llama_hub.tools.utils import create_schema_from_function
+from pydantic import BaseModel
+from llama_hub.tools.function_tool import FunctionTool
+
+
+class OnDemandToolLoaderTool(BaseTool):
```

Comment:
yes, naming is hard lol. definitely need something better than this.

Reply:
i'm thinking OnDemandQueryTool (or just OnDemandTool)

Reply:
What about just LoadAndQueryTool